rewrite/rewrite/python/format/spaces_visitor.py

```diff
@@ -408,6 +408,38 @@ def visit_expression_type_tree(self, expression_type_tree: ExpressionTypeTree, p
         ett = space_before(ett, False)
         return ett
 
+    def visit_comprehension_expression(self, comprehension_expression: ComprehensionExpression, p: P) -> J:
+        ce = cast(ComprehensionExpression, super().visit_comprehension_expression(comprehension_expression, p))
+
+        # Handle space before result this will depend on the style setting for the comprehension type.
+        if ce.kind in (ComprehensionExpression.Kind.LIST, ComprehensionExpression.Kind.GENERATOR):
```

Comment:
For generator expressions like `gen_exp = (x ** 2 for x in range(5))` I get the impression that none of the spaces styles actually applies. Instead PyCharm will just remove the spaces after the opening brace and before the closing brace (at least if it doesn't contain new lines or comments).

Reply:
I have modified them also to do the same.